diesel/src/sqlite/expression/functions.rs

```diff
@@ -892,4 +892,198 @@ extern "SQL" {
         j: J,
         path: Text,
     ) -> Nullable<Text>;
+
+    /// The json_extract(X,P1,P2,...) extracts and returns one or more values from the well-formed JSON at X.
+    ///
+    /// If only a single path P1 is provided, then the SQL datatype of the result is NULL for a JSON null,
+    /// INTEGER or REAL for a JSON numeric value, an INTEGER zero for a JSON false value, an INTEGER one for a JSON true value,
+    /// the dequoted text for a JSON string value, and a text representation for JSON object and array values.
```

Comment:
That means the signature as expressed below doesn't really match what is returned by the function. Given that is is essentially a runtime determined value I do not have any good recommendation how to make this work with diesels static checking.

I could imaging to expose just several variants of this function, so `json_extract_as_text`, `json_extract_as_integer`, `json_extract_as_float`  and `json_extract_as_json`  for the different return type variants and let the users chose the right one for their usecase. After all you can even convert any sql return value to any other type internally.